eras/alonzo/impl/src/Cardano/Ledger/Alonzo/Rules/Utxow.hs

```diff
@@ -227,11 +227,9 @@ hasExactSetOfRedeemers ::
   AlonzoScriptsNeeded era ->
   Test (AlonzoUtxowPredFailure era)
 hasExactSetOfRedeemers tx (ScriptsProvided scriptsProvided) (AlonzoScriptsNeeded scriptsNeeded) = do
-  let txBody = tx ^. bodyTxL
-      redeemersNeeded =
-        [ (rp, (sp, sh))
+  let redeemersNeeded =
+        [ (hoistPlutusPurpose toAsIndex sp, (hoistPlutusPurpose toAsItem sp, sh))
```

Comment:
Aha so, 
Before:  in order to check that the redeemers we were provided (using  indices) are exactly what we need, we would use AlonzoScriptsNeeded to get the items from ScriptPurpose for each tag an then `indexOf`  on it to find the index  (via the now defunct `rdptr` function).
But now: when constructing AlonzoScriptsNeeded, we pre-emptively index and associate it with the item, so we just need to get them from `getAlonzoScriptsNeeded` and compare them with keys of the redeemers provided in TxWits.
Is this about right?

Reply:
That's the perfect explanation of the change of how it worked before and how it'll work after this PR.